src/EditorFeatures/Core.Cocoa/Lsp/VSMacLspLoggerFactoryWrapper.cs

```diff
@@ -45,15 +45,40 @@ public VSMacLspLoggerWrapper(IVSMacLspLogger logger)
         _logger = logger;
     }
 
-    public void TraceError(string message) => _logger.TraceError(message);
+    public Task LogErrorAsync(string message, CancellationToken cancellationToken, params object[] @params)
```

Comment:
@CyrusNajmabadi and I talked a bit offline.  I'm leaning towards making these synchronous since the underlying API both of us use to log (loghub) will be ends up with synchronous `Log` APIs - https://sourceroslyn.io/#Microsoft.VisualStudio.LanguageServices/LanguageClient/LogHubLspLogger.cs,39

To write x-process logs someone would have to do a more complex impl with buffering but imho it seems reasonable (considering `ILogger` is also a sync API)